eduhelx_jupyterlab_prof/handlers.py

```diff
@@ -192,9 +194,18 @@ async def update_gitignore_master_notebook(self, assignment_name, master_noteboo
             gitignore_lines = []
             
         matches_gitignore = parse_gitignore(assignment_gitignore_path)
-        if not matches_gitignore(repo_root / assignment["directory_path"] / master_notebook_path):
+
+        if assignment["manual_grading"]:
+            # The master notebook is shared among students for manually graded assignments.
+            # We want to remove the new master notebook if possible (using a naive scan)
+            lines = [line for line in gitignore_lines if line.strip() != master_notebook_path.strip()]
```

Comment:
why not use `matches_gitignore()` here? is the functionality different in some way I'm not seeing?

Reply:
discussed offline